sql/sql_table.cc

```diff
@@ -4203,6 +4203,20 @@ mysql_prepare_create_table(THD *thd, HA_CREATE_INFO *create_info,
             DBUG_RETURN(TRUE);
           }
         }
+
+        /* Check that there's no repeating foreign key constraint names. */
+        List_iterator_fast<Key> key_it(alter_info->key_list);
+        Key *key;
+        while ((key= key_it++))
+        {
+          if (check->name.length == key->name.length &&
+              my_strcasecmp(system_charset_info,
+                            check->name.str, key->name.str) == 0 && key->type == Key::FOREIGN_KEY)
```

Comment:
InnoDB may already check this indirectly by defining a unique index on `SYS_FOREIGN.ID`. But, InnoDB does not store any `CHECK` constraints. And we’d like to move the FOREIGN KEY constraint metadata storage out of InnoDB.

I think that constraint names should be unique, no matter what the constraint type is.
Is the condition `&& key->type == KEY::FOREIGN_KEY` really necessary?

Reply:
Changed thanks @dr-m